packages/next/src/lib/metadata/generate/basic.tsx

```diff
@@ -157,6 +157,22 @@ export function FacebookMeta({
   ])
 }
 
+export function PinterestMeta({
+  pinterest,
+}: {
+  pinterest: ResolvedMetadata['pinterest']
+}) {
+  if (!pinterest) return null
+
+  const { richPin } = pinterest
+
+  return MetaFilter([
```

Comment:
let's change this to early assertions to avoid using MetaFilter for extra work

```
export function PinterestMeta({
  pinterest,
}: {
  pinterest: ResolvedMetadata['pinterest']
}) {
  if (!pinterest || !pinterest.richPin) return null

  const { richPin } = pinterest

  return <meta property="pinterest-rich-pin" content={richPin.toString()} />
}
```